std/stdio.d

```diff
@@ -4113,10 +4137,19 @@ private size_t readlnImpl(FILE* fps, ref char[] buf, dchar terminator = '\n')
         }
     }
 
-    char *lineptr = null;
-    size_t n = 0;
+    static char *lineptr = null;
+    static size_t n = 0;
+    scope(exit)
+    {
+        if (n > 128 * 1024)
+        {
+            // Bound memory used by readln
+            free(lineptr);
```

Comment:
I'm fine with leaking this until program exit, but valgrind would report this as leak.
Adding a `~static this() {}` might run into circular dependency issues though.